fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/SeriallyTaskScheduler.java

```diff
@@ -27,7 +27,7 @@ public void executeTasks(TaskContext context, Group group) {
         while (!tasks.empty()) {
             if (System.currentTimeMillis() >= endTime) {
                 // Should have at least one valid plan
-                if (!group.hasBestExpression(context.getRequiredProperty())) {
+                if (group != null && !group.hasBestExpression(context.getRequiredProperty())) {
```

Comment:
Seems this condition will always false when in logical rewrite phase. When timeout, the loop break but not throw exception may lead to unexpect behavior.

Reply:
Emm, we can get a complete tree if logical rewrite phase timeout, and the following process can be executed normally (copy tree into memo, and extract tree from memo)

Reply:
BYW, I will update the timeout check logic in next PR, the check only work on one task now, it's not what we expected

Reply:
Ok, I've encountered problems with endless rewriting sql because of rules conflict. if we didn't have a limit to stop this process, it maybe a problem.

Reply:
> Ok, I've encountered problems with endless rewriting sql because of rules conflict. if we didn't have a limit to stop this process, it maybe a problem.

If you meet dead loop, it should be the match pattern or the execution order of the rule is error,  we must fixed it, not stop the process by timeout